src/hooks/__tests__/useScanApi.test.ts

```diff
@@ -0,0 +1,445 @@
+import { renderHook } from '@testing-library/react';
+import axios from 'axios';
+import {
+  useCreateScanApi,
+  useDeleteScanApi,
+  useDownloadReportApi,
+  useGetScanJobsApi,
+  useRunScanApi,
+  useShowConnectionsApi
+} from '../useScanApi';
+
+describe('useShowConnectionsApi', () => {
+  let hookResult;
+
+  beforeEach(() => {
+    const hook = renderHook(() => useShowConnectionsApi());
+    hookResult = hook?.result?.current;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should attempt an api call to show connections', () => {
+    const { apiCall } = hookResult;
+    const spyAxios = jest.spyOn(axios, 'get');
+
+    apiCall({ id: 123, connection: { id: 456 } });
+    expect(spyAxios.mock.calls).toMatchSnapshot('apiCall');
+  });
+
+  it('should handle success while fetching connections', async () => {
+    const { showConnections } = hookResult;
+    jest.spyOn(axios, 'get').mockResolvedValue({
+      data: {
+        results: [
+          { status: 'success', name: 'Connection 1' },
+          { status: 'failed', name: 'Connection 2' },
+          { status: 'unreachable', name: 'Connection 3' }
+        ]
+      }
+    });
+
+    const response = await showConnections({ id: 123, connection: { id: 456 } });
+    expect(response).toMatchSnapshot('showConnections success');
+  });
```

Comment:
require: somehow we skipped `useShowConnections` and breaking out the function/method tests... doing this should help highlight an issue with `callbackSuccess` that was masked...

Looks like we just need to modify this `it` and add 2 more
```
it('should handle success while fetching connections', async () => {
    const { showConnections } = hookResult;
    jest.spyOn(axios, 'get').mockResolvedValue({});

    const response = await showConnections({ id: 123, connection: { id: 456 } });
    expect(response).toMatchSnapshot('showConnections success');
  });

...

  it('should process an API success response', () => {
    const { callbackSuccess } = hookResult;

    expect(
      callbackSuccess({
        data: {
          results: [
            { status: 'success', name: 'Connection 1' },
            { status: 'failed', name: 'Connection 2' },
            { status: 'unreachable', name: 'Connection 3' }
          ]
        }
      })
    ).toMatchSnapshot('callbackSuccess');
  });

  it('should process an API error response', async () => {
    const { callbackError } = hookResult;

    await expect(
      callbackError({
        response: {
          data: {
            message: 'Dolor sit'
          }
        }
      })
    ).rejects.toMatchSnapshot('callbackError');
  });
```